pinterest_export/url_parser.py

```
@@ -0,0 +1,55 @@
+
+    # Normalize: add scheme if missing
+    if url.startswith("/"):
+        url = "https://www.pinterest.com" + url
+    elif not url.startswith("http"):
+        url = "https://" + url
+
+    parsed = urlparse(url)
+
+    # Validate host
+    if parsed.hostname and "pinterest" not in parsed.hostname:
+        raise ValueError(f"Not a Pinterest URL: {raw}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)